New environment variable to change easily the publication mode (#470)
ros2/rmw_fastrtps: README.md

1. Placing your XML file in the running directory under the name `DEFAULT_FASTRTPS_PROFILES.xml`.
2. Setting environment variable `FASTRTPS_DEFAULT_PROFILES_FILE` to your XML file.

### Change publication mode

> Another way to change easily the publication mode is to use the environment variable `ROS_PUBLICATION_MODE`.

Reviewer: @JLBuenoLopez-eProsima I'm somewhat inclined to stick to `RMW` for a prefix: 
```suggestion
Another way to change easily the publication mode is to use the environment variable `RMW_PUBLICATION_MODE`.
```

Author: For the moment, I have changed the name to the one proposed by @ivanpauno. If later other rmw implementations start to use a similar variable to configure the publication mode we can consider updating this code.

1. Placing your XML file in the running directory under the name `DEFAULT_FASTRTPS_PROFILES.xml`.
2. Setting environment variable `FASTRTPS_DEFAULT_PROFILES_FILE` to your XML file.

### Change publication mode

Another way to change easily the publication mode is to use the environment variable `RMW_FASTRTPS_PUBLICATION_MODE`.
The admissible values are:
* `ASYNCHRONOUS`: asynchronous publication mode.

> This implies that when the publisher calls the write operation, the data is copied in a queue and control is returned to the user thread before the data is actually sent.
> The asynchronous thread is in charge of consuming the queue and send the data to any matched reader.

Reviewer 2: ```suggestion
Setting this mode implies that when the publisher invoques the write operation, the data is copied into a queue, a notification about the addition to the queue is performed, and control of the thread is returned to the user before the data is actually sent.
A background thread (asynchronous thread) is in turn in charge of consuming the queue and sending the data to every matched reader.
```

* `SYNCHRONOUS`: synchronous publication mode.

> This implies that the data is sent directly in the context of the user thread.

Reviewer 2: ```suggestion
Setting this mode implies that the data is sent directly within the context of the user thread.
This entails that any blocking call occurring during the write operation would block the user thread, thus preventing the application with continuing its operation.
It is important to note that this mode typically yields higher throughput rates at lower latencies, since the notification and context switching between threads is not present.
```

* `AUTO`: let Fast DDS select the publication mode. This implies using the publication mode set in the XML file or, failing that, the default value set in Fast DDS (which currently is set to `SYNCHRONOUS`).

> By default, `rmw_fastrtps` assumes this variable to be `ASYNCHRONOUS` and it is the publication mode in use if this environment variable is not set.

Reviewer 2: ```suggestion
By default, `rmw_fastrtps` assumes this variable to be `ASYNCHRONOUS`, which is the publication mode in use if this environment variable is not set.
```

The admissible values are:
* `ASYNCHRONOUS`: asynchronous publication mode.
Setting this mode implies that when the publisher invokes the write operation, the data is copied into a queue, a notification about the addition to the queue is performed, and control of the thread is returned to the user before the data is actually sent.
A background thread (asynchronous thread) is in turn in charge of consuming the queue and sending the data to every matched reader.
* `SYNCHRONOUS`: synchronous publication mode.
Setting this mode implies that the data is sent directly within the context of the user thread.
This entails that any blocking call occurring during the write operation would block the user thread, thus preventing the application with continuing its operation.
It is important to note that this mode typically yields higher throughput rates at lower latencies, since the notification and context switching between threads is not present.
* `AUTO`: let Fast DDS select the publication mode. This implies using the publication mode set in the XML file or, failing that, the default value set in Fast DDS (which currently is set to `SYNCHRONOUS`).

> By default, `rmw_fastrtps` assumes this variable to be `ASYNCHRONOUS`, which is the publication mode in use if this environment variable is not set.

Reviewer 3: Do you really mean "assumes" here? That is, is there an implicit assumption that the value is ASYNCHRONOUS? Or, does this mean that rmw_fastrtps sets the value, by default, to ASYNCHRONOUS? I suppose the code will clarify, but the description should be clear.

Reviewer 4: I think it must be "if `RMW_FASTRTPS_PUBLICATION_MODE` is not set, then `rmw_fastrtps_cpp` behaves as if it were set to `ASYNCHRONOUS`.

Author: Done in 2bb6f5fbb1e7f971ab941c944998d0ca20a23c88
